### DISCUSSION TITLE
Viewset.get_extra_action_url_map does not recognize namespaces

### DISCUSSION THREAD

**Discussion by backbohne** [encode/django-rest-framework#7816] [Ideas & Suggestions]
Seems `Viewset`s `get_extra_action_url_map` method does not recognize namespaces. I'm using different apps via Django URL namespaces ("<namespace>:<url name>") but the DRF action pull down menu will not provide it. My quick fix looks like:

```
    def get_extra_action_url_map(self):
        action_urls = OrderedDict()

        # exit early if `detail` has not been provided
        if self.detail is None:
            return action_urls

        # filter for the relevant extra actions
        actions = [
            action for action in self.get_extra_actions()
            if action.detail == self.detail
        ]
        for action in actions:
            try:
                url_name = '%s-%s' % (self.basename, action.url_name)
                namespace = self.request.resolver_match.namespace
                if namespace:
                    url_name = '%s:%s' % (namespace, url_name)
                url = reverse(url_name, self.args, self.kwargs,
                              request=self.request)
                view = self.__class__(**action.kwargs)
                action_urls[view.get_view_name()] = url
            except NoReverseMatch as exc:
                pass  # URL requires additional arguments, ignore

        return action_urls
```

which basically add these two lines:

```
                namespace = self.request.resolver_match.namespace
                if namespace:
                    url_name = '%s:%s' % (namespace, url_name)
```
The fix works fine for me since years, but it seems that nobody else has been stumbled about it before.

Regards
Frank

**Comment by gabn88** [comment]
Great fix! 
Was wondering for more then a year why I couldn't get the extra actions button working on the browsable api, while the extra actions were actually working :)

**Comment by Bergiu** [comment]
i think this fix should get integrated into drf.

**Comment by Brachamul** [comment]
Shouldn't this be marked as a bug rather than an idea/suggestion ?
In practice, if you set up extra actions on a viewset while using namespaced apps, the expected behaviour is broken.

**Reply by gabn88** [reply]
I agree. I now use a custom Viewset that includes the above code in almost all views because of this bug.

**Comment by rockstarr-programmerr** [comment]
I was debugging to see why my @action doesn't appear in browsable api, and found this too.
Agree that this should be considered a bug. Namespace urls by app is very common in Django and is the recommended way. If it causes DRF to not work as expected and for no obvious reason, then why is it not a bug?

**Comment by gabn88** [comment]
And the pull request is merged into master! 

See https://github.com/encode/django-rest-framework/pull/8598

When a new release is out I will update this post if I think about it.

**Reply by rockstarr-programmerr** [reply]
Thanks for getting this resolved! The out-of-the-box browsable api is that much more powerful now

**Reply by Brachamul** [reply]
Excellent ! Thank you for getting this fixed, great improvement.
